### PR TITLE
Added support for constraints copying

### PR DESCRIPTION
**Pull request by smamczak**
…czak@man.poznan.pl>	14 May 2020 at 15:01

Added support for constraints copying	6e3d9bc	Sebastian Mamczak <smamczak@man.poznan.pl>	14 May 2020 at 15:01

### CODE REVIEW

**Comment by smamczak** [comment]
Hi again, I added support for copying constraints

When you set up a view in ViewController you often want to set height or width (or both) or even aspect ratio of the view. Now these constraints are copied to UIView loaded with NibLoader

File: Sources/NibLoader.swift [hunk 1]
```diff
@@ -71,6 +73,28 @@ public struct IBNibLoader<NibLoadableView: NibLoadable> where NibLoadableView: U
...
     
+    private func copyConstraints(to nibView: UIView) {
+        let attributesToCopy: [NSLayoutConstraint.Attribute] = [.height, .width]
+        let constraintsToCopy = view.constraints.filter {
+            return attributesToCopy.contains($0.firstAttribute)
+        }
+        for constraint in constraintsToCopy {
+            var secondItem: Any? = nil
+            if attributesToCopy.contains(constraint.secondAttribute) {
+                secondItem = nibView
```
Comment (domasn) [inline Sources/NibLoader.swift:84]: It is not obvious wht this isn't the `contstraint.toItem` 🤔

Comment (smamczak) [reply Sources/NibLoader.swift:84]: For .height and .width constraint, there is only ```item``` (aka firstItem), but if you add aspect ratio from IB then the toItem (aka secondItem) should be set to the same view as firstItem - the secondAttribute is then .height or .width too

**Comment by smamczak** [comment]
@domasn alternatively I can change is to
```toItem: (constraint.secondItem == self ? nibView : nil)```
if it more obvious

**Comment by domasn** [comment]
I've reproduced the problem and managed a simpler fix based on your work: https://github.com/trafi/NibView/commit/da9e839049bcdf66b4ac4ca18f65d3db4fe76861
Note, that constraints copy wasn't necessary for `initWithFrame` that is only used during Interface Builder rendering and was already working just fine.

Thanks! 🙏 
[2.0.2](https://github.com/trafi/NibView/releases/tag/2.0.2) includes this 👌

**Pull request closed without merging**